src/easylink/step.py

```diff
@@ -737,111 +737,78 @@ def _validate_step_graph(
-                raise ValueError(
-                    f"{self.step_name} output slot {sm.parent_slot} maps to non-existent slot {sm.child_slot} on child node {sm.child_node}"
-                )
+            for sm in slot_mappings:
+                if sm.child_node not in self.step_graph.nodes:
+                    raise ValueError(
+                        f"{self.step_name} {slot_type} slot {sm.parent_slot} maps to non-existent child node {sm.child_node}"
+                    )
+                if sm.child_slot not in getattr(
+                    self.step_graph.nodes[sm.child_node]["step"], f"{slot_type}_slots"
+                ):
+                    raise ValueError(
+                        f"{self.step_name} {slot_type} slot {sm.parent_slot} maps to non-existent slot {sm.child_slot} on child node {sm.child_node}"
+                    )
 
     def _check_validators_are_consistent(self):
         """Check that if two input slots will receive the same data, they have the same validator.
 
         There are two versions of this to check: input slots that receive the same data because
-        they both are mapped from the same parent input slot, and input slots that receive the
+        one is mapped to the other by a slot mapping, and input slots that receive the
         same data because they both are at the receiving end of edges from the same output slot.
         """
-        child_input_slots_by_parent_input_slot = {}
+        # Check that input slots mapped to by our slot mappings have consistent validators
         for sm in self.slot_mappings["input"]:
-            if sm.parent_slot not in child_input_slots_by_parent_input_slot:
-                child_input_slots_by_parent_input_slot[sm.parent_slot] = []
-
-            child_input_slots_by_parent_input_slot[sm.parent_slot].append(
-                self.step_graph.nodes[sm.child_node]["step"].input_slots[sm.child_slot]
-            )
-
-        for (
-            parent_input_slot,
-            child_input_slots,
-        ) in child_input_slots_by_parent_input_slot.items():
-            first_validator = child_input_slots[0].validator
-            for child_input_slot in child_input_slots:
-                if child_input_slot.validator != first_validator:
-                    raise ValueError(
-                        f"Not all input slots mapped to from {self.step_name}'s {parent_input_slot} have the same validator"
-                    )
+            expected_validator = self.input_slots[sm.parent_slot].validator
+            child_input_slot = self.step_graph.nodes[sm.child_node]["step"].input_slots[
+                sm.child_slot
+            ]
+            if child_input_slot.validator != expected_validator:
+                raise ValueError(
+                    f"{sm.child_node}'s {sm.child_slot}, which is mapped from {self.step_name}'s {sm.parent_slot}, does not have the same validator"
```

Comment:
Nit: Shouldn't this be 
```suggestion
                    f"{sm.child_node}'s {sm.child_slot}, which is mapped from {self.step_name}'s parent slot {sm.parent_slot}, does not have the same validator"
```

Reply:
No, because "parent" (in the context of the SlotMapping) and "self" (the Hierarchical Step) here refer to the same thing. So `sm.parent_slot` is a regular slot on `self`, not some kind of special "parent slot."